packages/cli/src/util/deploy/create-git-meta.ts

```diff
@@ -6,22 +6,22 @@ import { exec } from 'child_process';
 import { GitMetadata } from '../../types';
 import { Output } from '../output';
 
-export function isDirty(directory: string): Promise<boolean> {
-  return new Promise((resolve, reject) => {
+export function isDirty(directory: string, output: Output): Promise<boolean> {
+  return new Promise(resolve => {
     exec('git status -s', { cwd: directory }, function (err, stdout, stderr) {
-      if (err) return reject(err);
-      if (stderr)
-        return reject(
-          new Error(
-            `Failed to determine if git repo has been modified: ${stderr.trim()}`
-          )
+      if (err) return resolve(false);
+      if (stderr) {
+        output.debug(
+          `Failed to determine if git repo has been modified: ${stderr.trim()}`
         );
+        return resolve(false);
+      }
       resolve(stdout.trim().length > 0);
     });
   });
 }
 
-function getLastCommit(directory: string): Promise<git.Commit> {
+function getLastCommit(directory: string): Promise<git.Commit | void> {
```

Comment:
```suggestion
function getLastCommit(directory: string): Promise<git.Commit> {
```

Is this return type still valid?